service/src/main/kotlin/fi/espoo/evaka/attendance/MobileUnitController.kt

```diff
@@ -12,33 +12,47 @@ import fi.espoo.evaka.shared.auth.AccessControlList
 import fi.espoo.evaka.shared.auth.AuthenticatedUser
 import fi.espoo.evaka.shared.auth.UserRole
 import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.db.mapColumn
+import fi.espoo.evaka.shared.db.mapRow
+import fi.espoo.evaka.shared.domain.EvakaClock
 import fi.espoo.evaka.shared.domain.NotFound
-import fi.espoo.evaka.shared.utils.dateNow
-import org.jdbi.v3.core.kotlin.mapTo
+import org.jdbi.v3.core.mapper.PropagateNull
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 import java.time.LocalDate
 
-// todo: move elsewhere?
 @RestController
-@RequestMapping("/mobile")
+@RequestMapping("/mobile/units")
 class MobileUnitController(
     private val acl: AccessControlList
 ) {
-    @GetMapping("/units/{unitId}")
+    @GetMapping
+    fun getUnitInfos(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        evakaClock: EvakaClock,
+        @RequestParam unitIds: List<DaycareId>
+    ): List<UnitInfo> {
+        Audit.UnitRead.log(targetId = unitIds)
+        unitIds.forEach { unitId -> acl.getRolesForUnit(user, unitId).requireOneOfRoles(UserRole.MOBILE) }
```

Comment:
This has the potential to translate to hundreds of queries each opening and closing their own connection, and should be rethinked.

Reply:
Moreover, I do not like the idea of adding attendance data to `UnitInfo` because the attendance numbers are not needed in most of the use cases, and the data might become stale quickly which could encourage someone to just refresh the unit data instead of using a more approriate solution.
Previously all unit data was refreshed on all page interactions and was included in `AttendanceResponse`. @Joosakur and I refactored long-lived unit data to a separate endpoint and context so that it can be kept in memory without constant need of refreshing.

Could the unit-level attendances just be fetched in a separate request when needed?